cmd/appsctl/provision_app.go

```diff
@@ -0,0 +1,146 @@
+// Copyright (c) 2019-present Mattermost, Inc. All Rights Reserved.
+// See License for license information.
+
+package main
+
+import (
+	"archive/zip"
+	"bytes"
+	"encoding/json"
+	"io"
+	"io/ioutil"
+	"os"
+	"strings"
+
+	"github.com/pkg/errors"
+
+	"github.com/mattermost/mattermost-plugin-apps/apps"
+	"github.com/mattermost/mattermost-plugin-apps/awsclient"
+)
+
+type functionInstallData struct {
+	zipFile io.Reader
+	name    string
+	handler string
+	runtime string
+}
+
+// ProvisionApp gets a release URL parses the release and creates an App in AWS
+// releaseURL should contain a zip with lambda functions' zip files and a `manifest.json`
+// ~/my_app.zip
+//  |-- manifest.json
+//  |-- my_nodejs_function.zip
+//      |-- index.js
+//      |-- node-modules
+//          |-- async
+//          |-- aws-sdk
+//  |-- my_python_function.zip
+//      |-- lambda_function.py
+//      |-- __pycache__
+//      |-- certifi/
+func ProvisionAppFromFile(awscli awsclient.Client, path string, shouldUpdate bool) error {
+	f, err := os.Open(path)
+	if err != nil {
+		return errors.Wrapf(err, "can't read file from  path %s", path)
+	}
+
+	b, err := ioutil.ReadAll(f)
+	if err != nil {
+		return errors.Wrap(err, "can't read file")
+	}
+
+	return ProvisionApp(awscli, b, shouldUpdate)
+}
+
+func ProvisionApp(awscli awsclient.Client, b []byte, shouldUpdate bool) error {
+	zipReader, err := zip.NewReader(bytes.NewReader(b), int64(len(b)))
+	if err != nil {
+		return errors.Wrap(err, "can't get zip reader")
+	}
+	bundleFunctions := []functionInstallData{}
+	var mani *apps.Manifest
+
+	// Read all the files from zip archive
+	for _, file := range zipReader.File {
+		switch {
+		case strings.HasSuffix(file.Name, "manifest.json"):
```

Comment:
move `manifest.json` to a const?

Reply:
This is the only place where it's used in a functional way, and it's in `appsctl`, tempted to leave it alone for now.